packages/base-dao/src/baseDAO.ts

```diff
@@ -335,6 +366,21 @@ export function createEntityDAO({ entityName, hooks, pluralizationFunction = plu
       };
       return dataProvider.getEntitiesConnection(entityName, transformedArgs);
     },
+    getRelatedEntityId: async (originalEntityId: string, relationEntityName: string) => {
+      const relation = await dataLoaderForSingleRelatedEntity.load({ originalEntityId, relationEntityName });
+      return relation && relation.id;
+    },
+    getRelatedEntity: async (originalEntityId: string, relationEntityName: string) => {
+      return dataLoaderForSingleRelatedEntity.load({ originalEntityId, relationEntityName });
+    },
+    getRelatedEntityIds: async (originalEntityId: string, relationEntityName: string, args?: any) => {
+      const relations = await dataLoaderForRelatedEntities.load({ originalEntityId, relationEntityName, args });
+      return relations && (Array.isArray(relations) ? _.without(relations, undefined).map(relation => relation.id) : relations.id);
```

Comment:
This should always return an array. Why is there an option that it returns a single id?

Reply:
i took this struct from the original function... i think it might be for the scenario of Nx using a join table, like e.g. billableItem <=> transaction.

Unfortunately we don't have this kind of join in the clou test entities so its a bit difficult to verify.